cpp/CMakeLists.txt

```diff
@@ -28,11 +28,28 @@ if (32BIT)
   set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -m32")
 endif ()
 
+# Options that can be passed to CMake using 'cmake -DKEY=VALUE'.
+option ("BUILD_GEOCODER" "Build the offline phone number geocoder" "ON")
+option ("USE_ALTERNATE_FORMATS" "Use alternate formats" "ON")
+option ("USE_BOOST" "Use Boost" "OFF")
+option ("USE_ICU_REGEXP" "Use ICU regexp engine" "ON")
+option ("USE_LITE_METADATA" "Use lite metadata" "OFF")
+option ("USE_RE2" "Use RE2" "OFF")
+option ("USE_STD_MAP" "Force the use of std::map" "OFF")
+option ("BUILD_STATIC_LIB" "Build static libraries" "ON")
+option ("BUILD_UNIT_TESTS" "Build unit tests (gtest sources are needed)" "OFF")
+
+if (ANDROID)
+  set (BUILD_GEOCODER OFF)
```

Comment:
gecoder is a binary for generating geo coding data. For cross-compilation builds, like Android, the binary should not be generated as it will not be run on Android.

Reply:
These changes are needed in order to allow libphonenumber to be integrated as subproject within another project using cmake.

Reply:
The compilation errors you are seeing are likely linked to your environment. Try to use instead a Docker container. On my side I have compiled only on macOS